Sources/MapboxNavigation/MapView.swift

```diff
@@ -174,33 +174,94 @@ extension MapView {
         
-        for layer in mapboxMap.style.allLayerIdentifiers.reversed() {
-            if !(layer.type.rawValue == "symbol") && !identifiers.contains(layer.id) {
-                let sourceLayer = mapboxMap.style.layerProperty(for: layer.id, property: "source-layer").value as? String
-                
-                if let sourceLayer = sourceLayer,
-                   sourceLayer.isEmpty {
-                    continue
-                }
-                
-                parentLayer = layer.id
+        var layerPosition: MapboxMaps.LayerPosition? = nil
+        var lowerLayers = Set<String>()
+        var upperLayers = Set<String>()
+        var foundLayer: Bool = false
+        var targetLayer: String? = nil
+        
+        if let indice = allAddedLayers.firstIndex(of: layerIdentifier) {
+            lowerLayers = Set(allAddedLayers.prefix(upTo: indice))
+            if allAddedLayers.indices.contains(indice + 1) {
+                upperLayers = Set(allAddedLayers.suffix(from: indice + 1))
+            }
+        }
+        
+        for layerInfo in mapboxMap.style.allLayerIdentifiers.reversed() {
+            if lowerLayers.contains(layerInfo.id) {
+                // find the topmost layer that should be below the layerIdentifier.
+                layerPosition = .above(layerInfo.id)
                 break
+            } else if upperLayers.contains(layerInfo.id) {
+                // find the bottommost layer that should be above the layerIdentifier.
+                layerPosition = .below(layerInfo.id)
+            } else if belowSymbol {
+                // find the topmost non symbol layer for layerIdentifier in belowSymbolLayers.
+                if !foundLayer,
+                   layerInfo.type.rawValue != "symbol",
+                   let sourceLayer = mapboxMap.style.layerProperty(for: layerInfo.id, property: "source-layer").value as? String,
+                   !sourceLayer.isEmpty {
+                    targetLayer = layerInfo.id
+                    foundLayer = true
+                }
+            } else {
+                // find the topmost layer for layerIdentifier in aboveSymbolLayers.
+                if !foundLayer,
+                   let sourceLayer = mapboxMap.style.layerProperty(for: layerInfo.id, property: "source-layer").value as? String,
+                   !sourceLayer.isEmpty {
+                    targetLayer = layerInfo.id
+                    foundLayer = true
+                }
             }
         }
         
-        return parentLayer
+        guard let targetLayer = targetLayer else { return layerPosition }
+        guard let layerPosition = layerPosition else { return .above(targetLayer) }
+        
+        if belowSymbol {
+            // for route line and arrow layers
+            if case let .below(sequenceLayer) = layerPosition, aboveSymbolLayers.contains(sequenceLayer) {
+                return .above(targetLayer)
```

Comment:
It’s kind of counterintuitive that the `.below(_:)` case leads to an `.above(_:)` position. In the future, someone unfamiliar with this code might think it’s a mistake and reverse it. Maybe explain what this code is doing in a comment.

Reply:
Thanks for the suggestion, I'll add comment for it.